src/main/js/ephox/imagetools/transformations/ImageTools.js

```diff
@@ -80,13 +82,13 @@ define("ephox/imagetools/transformations/ImageTools", [
 
   function resize(blob, w, h) {
     return Conversions.blobToImage(blob).then(function(image) {
-      var canvas = Canvas.create(w, h),
-        context = Canvas.get2dContext(canvas);
-
-      context.drawImage(image, 0, 0, w, h);
-      revokeImageUrl(image);
-
-      return Conversions.canvasToBlob(canvas, blob.type);
+      return ImageResizerWebgl.scale(image, w, h)
```

Comment:
Should resize take in an array of resizers since webgl might not be available or it might need to pick a different resizer depending on image size?

Like:
```
ImageTransformations.resize(blob, 100, 100, [
  ImageResizerWebgl,
  ImageResizerCanvas
]);
```

So the first one that produces a ImageResult from that list or resizers is the one that gets picked. So if the texture size for example is to small to fit the image for webgl it passes it on to the next canvas implementation. This way we can also change the priority of things or leave specific resizers out etc. There is a lot of implementations out there for resamling and all of them has pros/cons. Some algorithms work better for photos but not as good on non photos.